### PR TITLE
dragonball: fix for non-deterministic builds

### PR DESCRIPTION
**Pull request by alex-matei**
Fix for non-deterministic builds

Fixes: #7888

### CODE REVIEW

**Review by fidencio — APPROVED** [review]
lgtm, thanks @alex-matei!

**Comment by studychao** [comment]
Hi @alex-matei , in runtime-rs's built-in design, Dragonball is a Rust lib for runtime-rs and it doesn't have standalone binary. 
In Rust, the lib itself should not have Cargo.lock file, because it's up to the projects who use libs to define the Cargo lock. In this way, it is runtime-rs 's responsibility to keep the Cargo.lock.

So I'm curious why to introduce Cargo.lock in this PR.

File: src/dragonball/.gitignore [hunk 1]
```diff
@@ -1,3 +1,2 @@
 target
-Cargo.lock
```
Comment (studychao) [inline src/dragonball/.gitignore:2]: Ideally, we don't need lock file for a Rust lib.

**Review by studychao — CHANGES_REQUESTED** [review]

**Comment by alex-matei** [comment]
@studychao Dragonball is defined as a standalone component in the Makefile https://github.com/kata-containers/kata-containers/blob/46e73cf7a230f861feb4565d0d5d50130e25236f/Makefile#L11 . When running `make check`, which the `static-checks` job runs on each PR, the target `check` and its dependency, `build`, will run.

**Comment by studychao** [comment]
@alex-matei @fidencio ok, that makes sense.

Ideally, Dragonball is a lib to runtime-rs and it should not have a Cargo.lock itself. But we surely need to keep Dragonball working when we only want to build Dragonball alone. Introduce a Cargo.lock file could help with that.

LGTM. Thanks!

**Review by studychao — APPROVED** [review]

**Review by GabyCT — APPROVED** [review]

**Pull request merged**